examples/multiturn_demo/agent_loop/mcp_agent_loop.py

```diff
@@ -0,0 +1,201 @@
+# Copyright 2025 The RLinf Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+import copy
+import json
+import random
+import time
+from typing import Any
+from uuid import uuid4
+
+from omegaconf import DictConfig
+
+from rlinf.data.tool_call.tool_io_struct import (
+    ToolChannelRequest,
+    ToolChannelResponse,
+    ToolRequest,
+    ToolResponse,
+)
+from rlinf.utils.placement import ModelParallelComponentPlacement
+from rlinf.workers.agent.agent_loop import AgentLoopOutput, AgentLoopWorkerBase
+
+
+class MCPAgentLoopWorker(AgentLoopWorkerBase):
+    """Simple tool agent loop that can interact with tools."""
+
+    def __init__(
+        self,
+        cfg: DictConfig,
+        placement: ModelParallelComponentPlacement,
+    ):
+        super().__init__(cfg, placement)
+
+    def generate_context_create(self) -> dict[str, Any]:
+        return {
+            "tool_session_ids": {},
+        }
```

Comment:
it is a little strange that you wrap one more dict out of session ids. I guess you want to deal with more keys in future. But I think if there are more keys in future, it is better to use data class, in which case the code should be changed anyway.